README.md

```diff
@@ -2,19 +2,56 @@
 
 ## Description
 
-TODO: Describe your charm in a few paragraphs of Markdown
+This charm is used to configure an Ory Kratos charm to use an external provider.
 
 ## Usage
 
-TODO: Provide high-level usage, such as required config or relations
+### Client registration
 
-## Relations
+Before deploying this charm you should go and register an oidc client with the provider you wish to use. Instructions for
+registering a client for some well known provider can be found below. Instructions for a larger list of providers can be found at the
+Ory Kratos [docs](https://www.ory.sh/docs/kratos/social-signin/overview)
 
-TODO: Provide any relations which are provided or required by your charm
+Note that after registering a client you will need to provide a redirect_uri to the provider, you will get the redirect_uri
+after deploying the integrator.
```

Comment:
```suggestion
Note that after registering a client you will need to provide a redirect_uri to the provider. It can be fetched
once the integrator charm is deployed.
```